filter_plugins/redis.py

```diff
@@ -0,0 +1,68 @@
+class FilterModule(object):
+    def filters(self):
+        return {
+            'redis_master_address': self.redis_master_address,
+            'redis_master_port': self.redis_master_port,
+        }
+
+    def redis_master_address(self, fact, redis_type='redis', default_address=None, namespace='OPENIO', service=None, monitor_name='OPENIO-master-1'):
+        try:
+            if redis_type == 'redis':
+                return self._master_address_redis(fact, default_address=default_address, namespace=namespace, service=service)
+            elif redis_type == 'redissentinel':
+                return self._master_address_sentinel(fact, default_address=default_address, namespace=namespace, service=service, monitor_name=monitor_name)
+            else:
+                return default_address
+        except Exception as e:
+            return default_address
+
+    def _master_address_redis(self, fact, default_address=None, namespace='OPENIO', service=None):
+        if 'instances' in fact:
+            for instance in fact['instances']:
```

Comment:
Prefer using `.get(key, default)` to remove one level of indentation.
`fact.get('instances', [])`